client.go

```diff
@@ -115,16 +140,38 @@ func (c *Client) handleResponse(apiResponse *http.Response, response interface{}
 		var errorResponse ErrorResponse
 		err := json.Unmarshal(body, &errorResponse)
 		if err != nil {
-			return fmt.Errorf("api error(%v): no body", apiResponse.StatusCode)
+			return NewMapboxError(apiResponse.StatusCode, "")
 		}
 
-		return fmt.Errorf("api error(%v): %v", apiResponse.StatusCode, errorResponse.Message)
+		// If rate limited, hold off till the next X-Rate-Limit-Reset
+		if apiResponse.StatusCode == 429 {
```

Comment:
I wanted to see an explicit example of what Mapbox returned for rate limits to verify their docs, but unfortunately I couldn't, so no e2e test on this has been done.

* I didn't want to test with spamming geocoding or directions matrix, in case the limits are account limits
* The API I did use (tokens API since it is supposedly only 100/min) I was never able to get throttled on. Maybe our account is flagged already to have higher limits??